### PR TITLE
Add togglable advanced gas controls on send and confirm screens

### PR DESCRIPTION
**Pull request by danjm**
Fixes https://github.com/MetaMask/metamask-extension/issues/5931

Demo:

peek 2019-02-05 13-50

cc @whymarrh @cjeria

### CODE REVIEW

**Comment by danjm** [comment]
I still need to update tests, but code should be good to review

**Comment by cjeria** [comment]
@danjm Can you add more padding above the "advanced options" button link? Like 16px. It's so tight up against the controls above it and could use more breathing room.

File: ui/app/components/gas-customization/advanced-gas-inputs/advanced-gas-inputs.component.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
+import React, { Component } from 'react'
```
Comment (danjm) [inline ui/app/components/gas-customization/advanced-gas-inputs/advanced-gas-inputs.component.js]: Note that this code is somewhat of a subset of code already in `metamask-extension/ui/app/components/gas-customization/gas-modal-page-container/advanced-tab-content/advanced-tab-content.component.js`

I will probably remove corresponding code from that component in a future task

File: ui/app/components/gas-customization/advanced-gas-inputs/advanced-gas-inputs.component.js [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    customGasLimit: PropTypes.number,
+    insufficientBalance: PropTypes.bool,
+    customPriceIsSafe: PropTypes.bool,
+    isSpeedUp: PropTypes.bool,
+  }
+
+  constructor (props) {
+    super(props)
+
+    this.debouncedGasLimitReset = debounce((dVal) => {
```
Comment (whymarrh) [inline ui/app/components/gas-customization/advanced-gas-inputs/advanced-gas-inputs.component.js]: Can we move these properties out of the constructor and use the property syntax? (And then even remove the empty constructor.)

**Review by whymarrh — CHANGES_REQUESTED** [review]

**Comment by danjm** [comment]
user experience when confirming from a dapp:

peek 2019-02-05 14-47

File: ui/app/components/gas-customization/advanced-gas-inputs/advanced-gas-inputs.component.js (hunk 2 above)
Comment (danjm) [reply ui/app/components/gas-customization/advanced-gas-inputs/advanced-gas-inputs.component.js]: Done in 0112680d6

**Comment by cjeria** [comment]
@danjm Can you move it down like 10px-16px more? Sry, super nitpic i know

**Comment by danjm** [comment]
@cjeria 

screenshot from 2019-02-05 16-22-48

**Review by whymarrh — APPROVED** [review]

**Pull request merged**